manifests/extract.pp

```diff
@@ -78,21 +79,27 @@
     $strip_opt = ''
   }
 
+  if $untar_opts != '' {
+      $untar_opts_real = " ${untar_opts}"
+  } else {
+      $untar_opts_real = ''
+  }
+
   case $name {
     /.tar$/: {
-      $command = "tar xf ${source_path}${strip_opt}"
+      $command = "tar xf ${source_path}${strip_opt}${untar_opts_real}"
     }
 
     /(.tgz|.tar.gz)$/: {
       if $::osfamily == 'Solaris' {
         $command = "gunzip -dc < ${source_path} | tar xf - "
```

Comment:
Should this also use the untar opts?

Reply:
I'm uncertain about that... there seems to be special handling around Solaris there, however I do not have any Solaris systems available to test with, so I cannot confirm if adding more options to the end of the command would break things.

Reply:
Same here, but the defaults are empty so it shouldn't bother users. POSIX also specifies various options so I'm assuming it must accept some.

Reply:
Good point, added another commit with this.